### DISCUSSION TITLE
Storm Watch Indicator?

### DISCUSSION THREAD

**Discussion by JohnJ9ml** [jasonacox/pypowerwall#126] [Q&A]
I've been looking around for some sort of indicator in the API responses that shows that a Storm Watch event has been triggered. It might be more significant to me right now only because I'm in that nether period where I'm between final approval and PTO, so I'm only allowed to  charge from the grid in preparation for an anticipated storm.

I'd like to be able to have a clue to identify the power used purely to charge the batteries in anticipation of this, but I haven't spotted an alert or flag thus far in any of the API responses. This is what led me to look at the Vitals output, but I'm not finding anything there either so far. Can anyone point me at something I'm missing?

**Comment by JohnJ9ml** [comment]
Okay, I might just now have spotted it:

I just picked up a system alert RampRateLimited. Perhaps this is the indicator.

LATER:  Okay, no, that's not it, but I'm going to stop guessing.

**Reply by jasonacox** [reply]
Hi @JohnJ9ml - That would be interesting to capture. Please continue your investigation for us!

If it shows up anywhere, I suspect it could be in one of the TEDAPI payloads - if you are running the proxy, you could try:
* http://localhost:8675/tedapi/config 
*  http://localhost:8675/tedapi/status 
*  http://localhost:8675/tedapi/battery

Or using python:

```python
import pypowerwall

host = "192.168.91.1"
gw_pw = "ABCDEFGHIJ"
pw = pypowerwall.Powerwall(host, "", "", "", gw_pwd=gw_pwd, auto_select=True)

# Pick one
json.dumps(pw.tedapi.get_config())
json.dumps(pw.tedapi.get_status())
json.dumps(pw.tedapi.get_components())
json.dumps(pw.tedapi.get_battery_blocks())
json.dumps(pw.tedapi.get_device_controller())
```

**Reply by JohnJ9ml** [reply]
I'll continue to work on this.

I notice that the tedapi calls can find my battery blocks, but the system_status output doesn't list them.

**Reply by jasonacox** [reply]
Thanks @JohnJ9ml  - and yes, please note any missing bits. If you are interested, also feel free to submit a pull request if you are willing to code the fixes/additions. At a minimum, we can open an issue for the missing battery blocks (would be good to have the example data).

**Comment by JohnJ9ml** [comment]
Continuing on with information I've turned up:

I had a call with Tesla customer service regarding an unrelated issue I've discovered processing the data here. In my discussions with them, I asked if there was an Alert that gets raised when a storm watch event triggers the system to fully charge the batteries. It took several tries to explain what I was looking for and where I was looking for it, but eventually an answer cam back that there currently is no such Alert in the system.

I've been looking about, and the only Alert that stands out that shows up is a RampRateLimited. Given the naming of it, I almost think this is just throttling the rate of charge. I say that because we're finally having rain again (very dry January for the lawn, not for the adults in the house :D ), and there is little to no production coming off the roof. Rather than pay for prime time from the grid, I've been manually pushing the reserve up to 80%, then pushing it back down to 20% right before 3pm. When I increase the reserve, the system works to get me there pronto:  pulling 12-14Kw to charge as quickly as possible. For the storm watch events, it charges below 10Kw, and more around the 8Kw range.

Just leaving bread crumbs as I continue to try to find something telltale that exposes a storm watch event is happening.

**Reply by jasonacox** [reply]
Thanks, John! The Alerts we are internal system signals that are not exposed to system owners, so it isn't surprising that Tesla customer service would not know about them. 

On the manual charge to avoid prime time, I've done the same thing, but use an automated script that works year around to adjust based on month, cloud coverage, and time (e.g. peak): https://github.com/jasonacox/pypowerwall/blob/main/tools/cron.sh - I recently updated it to watch for weather alerts  (but not charging during peak and also aware of fire conditions that cause power outages). It depends on our local NWS alerts so I haven't uploaded that change.
